packages/xrpc-server/src/server.ts

```diff
@@ -481,30 +484,27 @@ export class Server {
   }
 }
 
-function isHandlerSuccess(v: HandlerOutput): v is HandlerSuccess {
-  return handlerSuccess.safeParse(v).success
-}
-
-function isHandlerPipeThrough(v: HandlerOutput): v is HandlerPipeThrough {
-  if (v === null || typeof v !== 'object') {
-    return false
-  }
-  if (!isString(v['encoding']) || !(v['buffer'] instanceof ArrayBuffer)) {
-    return false
-  }
-  if (v['headers'] !== undefined) {
-    if (v['headers'] === null || typeof v['headers'] !== 'object') {
-      return false
-    }
-    if (!Object.values(v['headers']).every(isString)) {
-      return false
+const createValidator =
```

Comment:
I think we avoided zod here because it can be kinda heavy duty for validation of stuff that is often invalid. I know it used to handle this by throwing and catching errors which can be rough on Node so we tried to keep it out of the happy/critical path of our server stuff. 

But I also have some vague memory that they fixed this & made it faster? Any idea? Or any idea how these compare?

Reply:
You are right, validating here is way overkill. Since we know the input is of type `HandlerOutput`, all we need is to discriminate based on that. We can even make some of the checks we currently do disappear !